design/design-doc-observe-only-resources.md

```diff
@@ -0,0 +1,293 @@
+
+| Deletion Policy | Management Policy | Should Observe? | Should Create? | Should Update? | Should Delete? |
+| --- | --- | --- | --- | --- | --- |
+| Delete | Full | Yes | Yes | Yes | Yes |
+| Orphan | ObserveCreateUpdate | Yes | Yes | Yes | No |
+| Delete | ObserveOnly | Yes | No | No | Conflict (No) |
+| Orphan | Full | Yes | Yes | Yes | Conflict (No) |
+| Delete | ObserveCreateUpdate | Yes | Yes | Yes | Conflict (No) |
+| Orphan | ObserveOnly | Yes | No | No | No |
+
+For conflicting cases, we will decide based on the non-default configuration and not delete the external resource.
+This way, we will also err on the side of caution by leaving the actual resource. However, this comes at the cost of
+some possible confusion for users focusing on only one of the policies.
+
+Another solution could be simply throwing an error and preventing reconciliation during conflict. This would be more
+explicit but would require some manual actions and degraded UX for the usage of the feature, for example:
+
+- Creating an ObserveOnly resource will require both setting `managementPolicy` to `ObserveOnly` and `deletionPolicy`
+to `Orphan` .
+- If there are existing resources with `deletionPolicy: Orphan` when the feature is enabled, they will start failing to
+reconcile until their `managementPolicy`’s updated to `ObserveCreateUpdate`.
+
+#### Schema Changes
+
+The proposed approach here involves utilizing the same CR, hence schema, for both managing and observing resources.
+The caveat here is that some fields are required for creating the resources but not for observing. This means that it
+won’t be possible to create an observe-only resource without providing a value for any required fields, because
+the Kubernetes API checks for the presence of these fields before allowing the CR to be created.
+
+> Please note this is already the case with [importing existing managed resources] today.
+
+We will fix this by leveraging the [Common Expression Language (CEL)], which was graduated to beta (i.e. enabled by default)
+as of Kubernetes 1.25. See the following diff for the required changes we need for making [CIDRBlock] parameter of
+AWS VPC required only if not `ObserveOnly`:
+
+```diff
+		// CIDRBlock is the IPv4 network range for the VPC, in CIDR notation. For
+        // example, 10.0.0.0/16.
+-       // +kubebuilder:validation:Required
+        // +immutable
+-       CIDRBlock string `json:"cidrBlock"`
++       CIDRBlock *string `json:"cidrBlock,omitempty"`
+
+        // The IPv6 CIDR block from the IPv6 address pool. You must also specify Ipv6Pool
+        // in the request. To let Amazon choose the IPv6 CIDR block for you, omit this
+@@ -170,6 +169,7 @@ type VPC struct {
+        metav1.TypeMeta   `json:",inline"`
+        metav1.ObjectMeta `json:"metadata,omitempty"`
+
++       // +kubebuilder:validation:XValidation:rule="self.managementPolicy == 'ObserveOnly' || has(self.forProvider.cidrBlock)",mess
```

Comment:
I'm excited to potentially apply this pattern to generic cross-resource references, which also need more advanced logic to specify whether a field is required or not per https://github.com/crossplane/crossplane/issues/1770#issuecomment-1227805874 (e.g. "required unless its computed by a reference").